lib/internal/encoding.js

```diff
@@ -319,6 +323,15 @@ class TextEncoder {
     return encodeUtf8String(`${input}`);
   }
 
+  encodeInto(src, dest) {
+    validateEncoder(this);
+    validateString(src, 'src');
+    const items = [...src];
```

Comment:
Why are we spreading the tokens here? `encodeInto` is designed to be more efficient than `encode` but this (along with all the `Get()` calls in C++) would slow down the whole implementation quite a lot. It would be at least faster if we put the tokens into a TypedArray, or if we have to end up with calling `Object::Get` (neither `String` or `Array` has overload of that), we could just call it on the string instead of an array.

[From the spec](https://encoding.spec.whatwg.org/#dom-textencoder-encodeinto) I think it should be safe to do what `EncodeUtf8String()` does with the twist that we get the buffer from user land and return the return value of `WriteUtf8()` as `written`, the `Utf8Length()` of the original string as `read`, and handle the special case when the buffer is not big enough? Not sure if that fails any edge cases, but I am curious why this has to be implemented with the spread and actually iterating over the sequence.

Reply:
/ping @AtticusYang

Reply:
@joyeecheung Do you want to make changes and push directly to the branch in this PR? I'm interested in making sure this doesn't stall.

Reply:
I don’t think this PR needs to stall over this – it would be nice to have something (more) performant, but that shouldn’t be a blocker for landing this

Reply:
@addaleax I would normally agree but for `encodeInto` whose purpose is to avoid an extra copy, an implementation that costs more memory than a TypedArray copy seems misleading - e.g. users could calculate their memory usage with the expectation that this is not going to cost as much memory as the doing a `encode`, or even more, but then run into unexpected OOM (and worse, since the copy even uses V8 heap memory). That looks like a bug to me.

I would be more OK if the memory cost is noted in the docs (which this PR needs) but it would be quite weird to say `encodeInto` costs more memory than doing `encode` IMO.

Reply:
> Do you want to make changes and push directly to the branch in this PR? I'm interested in making sure this doesn't stall.

@Trott I could if there is not a particular reason that this has to be implemented with a copy in array of strings, maybe @AtticusYang has a very valid reason that eluded me, but there is not a reply yet so I do not know for sure.

Reply:
@joyeecheung @addaleax @Trott,  sorry,  I was too busy to see your feedback. @joyeecheung I think your advice is great,  I will take your advice to implement a new encodeInto in the weekends.

Reply:
@joyeecheung  It's more convenient to calculate the `read` and written bytes with splitting the src string into array. But it's less efficient. Thanks for your advice. @joyeecheung @addaleax @Trott  thanks for your kind help.